### PR TITLE
Add suffixes to typos that start with "ar"

### PR DESCRIPTION
**Pull request by int-y1**
I put `arange` and `aranges` in dictionary_code

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -4407,6 +4494,7 @@ arguemnt->argument
 arguemnts->arguments
 arguemtn->argument
 arguemtns->arguments
+arguent->argument
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:4497]: Is it worth adding `argumentive` and `argumentation`/`argumentations` typos, perhaps even `argumentatively`?
```
arguentation->argumentation
arguentations->argumentations
arguentatively->argumentatively
arguentive->argumentive
```

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -4333,51 +4361,110 @@ archine->archive
...
 archtiectures->architectures
+archtitect->architect
+archtitects->architects
+archtitectural->architectural
+archtitecturally->architecturally
 archtitecture->architecture
 archtitectures->architectures
 archtype->archetype
 archtypes->archetypes
+archve->archive
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary.txt:4440]: Perhaps add the `archival` adjective? 
⚠ In some cases, relevant typos could be from `archrival`.

File: codespell_lib/data/dictionary_code.txt [hunk 3]
```diff
@@ -2,6 +2,8 @@ agrv->argv
 alloced->allocated
 amin->main
 apoint->appoint
+arange->arrange, a range,
+aranges->arranges
```
Comment (DimitriPapadopoulos) [inline codespell_lib/data/dictionary_code.txt:6]: I guess `arange` is for `numpy.arange`, but `aranges`?

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (int-y1) [reply codespell_lib/data/dictionary.txt:4497]: Not in this PR. These uncommon words need a lot of new entries.

File: codespell_lib/data/dictionary_code.txt (hunk 3 above)
Comment (int-y1) [reply codespell_lib/data/dictionary_code.txt:6]: On github, most uses of `aranges` were related to DWARF. Should `aranges` go in dictionary_code or dictionary?

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (int-y1) [reply codespell_lib/data/dictionary.txt:4440]: Done. I didn't see any typos from `archrival`.

File: codespell_lib/data/dictionary_code.txt (hunk 3 above)
Comment (DimitriPapadopoulos) [reply codespell_lib/data/dictionary_code.txt:6]: Good question. I don't really know. I guess references to such DWARF details are rare enough to move `aranges` to the main dictionary. We can move it back to the code dictionary later on, if there are too many complaints.

**Review by DimitriPapadopoulos — APPROVED** [review]

**Pull request merged**